ffs/scheduler/scheduler.go

```diff
@@ -391,15 +436,30 @@ func (s *Scheduler) executeQueuedJob(j ffs.Job) {
 		s.l.Log(ctx, a.Cfg.Cid, "Job %s execution failed.", j.ID)
 		return
 	}
+	// Save whatever stored information was completely/partially
+	// done in execution.
 	if err := s.cis.Put(info); err != nil {
 		log.Errorf("saving cid info to store: %s", err)
 	}
-	if err := s.js.Finalize(j.ID, ffs.Success, nil, dealErrors); err != nil {
+
+	finalStatus := ffs.Success
+	// Detect if user-cancelation was triggered
+	select {
+	case <-cancelChan:
+		finalStatus = ffs.Canceled
+	default:
+	}
```

Comment:
So the Job is terminating. Is this because it finished as expected, or because someone canceled it?